core/components/ui/carousel/carousel.tsx

```diff
@@ -1,50 +1,42 @@
 import useEmblaCarousel, { UseEmblaCarouselType } from 'embla-carousel-react';
 import { ArrowLeft, ArrowRight } from 'lucide-react';
-import {
-  ComponentPropsWithoutRef,
-  ReactNode,
-  useCallback,
-  useEffect,
-  useId,
-  useMemo,
-  useState,
-} from 'react';
+import { ReactNode, useCallback, useEffect, useId, useMemo, useState } from 'react';
 
 import { cn } from '~/lib/utils';
 
 type CarouselApi = UseEmblaCarouselType[1];
 
-interface Props extends ComponentPropsWithoutRef<'div'> {
-  items: ReactNode[];
+interface Props {
+  className?: string;
+  products: ReactNode[];
   title: string;
-  pageSize?: 2 | 3 | 4;
```

Comment:
Is there a reason for removing this prop?

Reply:
It was added from a suggestion that this would be required for new components, but it is not the case, and our existing use case has no need for other page sizes, but I could keep for now if that is what we prefer.

Reply:
Seems more like a question we should ask our merchants/partners 🤔 

I don't really have a strong opinion here though, just curious what the reasoning was.

Reply:
I just added this a couple of weeks ago, I don't feel it will have a lot of impact by removing it.

Reply:
I added it back.